app/src/main/java/com/example/triptracker/model/repository/ItineraryRepository.kt

```diff
@@ -197,6 +207,108 @@ open class ItineraryRepository {
         .addOnFailureListener { e -> Log.e(TAG, "Error removing itinerary", e) }
   }
 
+  /**
+   * Increment a field in the itinerary document
+   *
+   * @param itineraryId Id of the itinerary document
+   * @param fieldName Name of the field to increment
+   * @return Unit
+   */
+  fun incrementField(itineraryId: String, fieldName: String) {
+    val itineraryRef = itineraryDb.document(itineraryId)
+    db.runTransaction { transaction ->
+          val snapshot = transaction.get(itineraryRef)
+          val currentCount = snapshot.getLong(fieldName) ?: 0
+          transaction.update(itineraryRef, fieldName, currentCount + 1)
+        }
+        .addOnSuccessListener { Log.d(TAG, "Field incremented successfully") }
+        .addOnFailureListener { e -> Log.e(TAG, "Error incrementing field", e) }
+  }
```

Comment:
Would it be a good idea to create a enum class with the different field names that can be passed ? This way we could ensure that the fieldName passed is always a defined field name and not another random string

Reply:
I didn't in the end, only the fields useful/related to the Home Screen